src/components/layout/Accordion/AccordionItem.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import * as styles from './AccordionItem.scss';
+import { Container } from '../../modules/Container/Container';
+import ILocalContainerProps from '../../../common/structures/ILocalContainerProps';
+import AnimateHeight from 'react-animate-height';
+import { CaretIcon, CheckmarkMixedIcon } from '../../icons/Icons';
+
+const animationStateClasses = {
+	animating: 'AccordionItem_Content__animating',
+	animatingUp: 'AccordionItem_Content__animatingUp',
+	animatingDown: 'AccordionItem_Content__animatingDown',
+	animatingToHeightZero: 'AccordionItem_Content__animatingToHeightZero',
+	animatingToHeightAuto: 'AccordionItem_Content__animatingToHeightAuto',
+	animatingToHeightSpecific: 'AccordionItem_Content__animatingToHeightSpecific',
+	static: 'AccordionItem_Content__static',
+	staticHeightZero: 'AccordionItem_Content__staticHeightZero',
+	staticHeightAuto: 'AccordionItem_Content__staticHeightAuto',
+	staticHeightSpecific: 'AccordionItem_Content__staticHeightSpecific',
+}
+
+interface PropsBase extends ILocalContainerProps {
+	children: React.ReactNode & {props?: any[]} | null;
+	clickArea?: 'icon' | 'all';
+	icon?: boolean | React.ReactNode;
+	itemId?: string;
+	onToggle?: (id: PropsControlled['itemId'] | undefined) => void;
+}
+
+interface PropsControlled extends PropsBase {
+	opened: boolean;
+}
+
+const AccordionItemControlled: React.FC<PropsControlled> = ({
+	children,
+	className,
+	clickArea = 'icon',
+	container,
+	icon = true,
+	id,
+	itemId,
+	onToggle,
+	opened,
+	style,
+}) => {
+	const onClick = (_: React.MouseEvent<HTMLElement>) => {
+		if (clickArea !== 'all') {
+			// accordion callback
+			onToggle?.(itemId);
+		}
+	};
+	const onClickSummary = (event: React.MouseEvent<HTMLElement>) => {
+		if (clickArea === 'all') {
+			// accordion callback
+			onToggle?.(itemId);
+		}
+	};
+
+	const TagSummary: any = clickArea === 'all' ? 'button' : 'div';
+
+	return (
+		<Container {...container}>
+			<div
+				className={classnames(
+					styles.AccordionItem,
+					'AccordionItem',
+					className,
+				)}
+				id={id}
+				style={style}
+			>
+				<>
+					{/* need to unwrap children because of potential extra 'child' wrapper (plus this works with fragments) */}
+					{ React.Children.map(children?.props?.children || children, (child: React.ReactNode, index: number) => {
```

Comment:
Wouldn't this be equivalent?
```suggestion
					{ React.Children.map(children, (child: React.ReactNode, index: number) => {
```

Reply:
Apparently not. I debugged this and given the unwrapping of the the intended parent element (if any), this actually does something different.